drivers/volume/portworx/portworx.go

```diff
@@ -506,35 +506,76 @@ func (p *portworx) inspectVolume(volDriver volume.VolumeDriver, volumeID string)
 			Type: "Volume",
 		}
 	}
+	return p.constructStorkVolume(vols[0]), nil
+}
 
+func (p *portworx) constructStorkVolume(vol *api.Volume) *storkvolume.Info {
 	info := &storkvolume.Info{}
-	info.VolumeID = vols[0].Id
+	info.VolumeID = vol.Id
 
-	info.VolumeName = vols[0].Locator.Name
-	for _, rset := range vols[0].ReplicaSets {
+	info.VolumeName = vol.Locator.Name
+	for _, rset := range vol.ReplicaSets {
 		info.DataNodes = append(info.DataNodes, rset.Nodes...)
 	}
-	if vols[0].Source != nil {
-		info.ParentID = vols[0].Source.Parent
+	if vol.Source != nil {
+		info.ParentID = vol.Source.Parent
 	}
 
 	info.Labels = make(map[string]string)
-	for k, v := range vols[0].Spec.GetVolumeLabels() {
+	for k, v := range vol.Spec.GetVolumeLabels() {
 		info.Labels[k] = v
 	}
 
-	for k, v := range vols[0].Locator.GetVolumeLabels() {
+	for k, v := range vol.Locator.GetVolumeLabels() {
 		info.Labels[k] = v
 	}
-	info.NeedsAntiHyperconvergence = vols[0].Spec.Sharedv4 && vols[0].Spec.Sharedv4ServiceSpec != nil
+	info.NeedsAntiHyperconvergence = vol.Spec.Sharedv4 && vol.Spec.Sharedv4ServiceSpec != nil
 
 	info.WindowsVolume = p.volumePrefersWindowsNodes(info)
 
-	info.VolumeSourceRef = vols[0]
+	info.VolumeSourceRef = vol
+
+	info.AttachedOn = vol.AttachedOn
 
-	info.AttachedOn = vols[0].AttachedOn
+	return info
 
-	return info, nil
+}
+
+func (p *portworx) enumerateVolumes(volumeNameList []string) (map[string]*storkvolume.Info, error) {
+	if !p.initDone {
+		if err := p.initPortworxClients(); err != nil {
+			return nil, err
+		}
+	}
+
+	volDriver, err := p.getAdminVolDriver()
+	if err != nil {
+		return nil, err
+	}
+	volMap := make(map[string]*storkvolume.Info)
+	vols, err := volDriver.Enumerate(
+		&api.VolumeLocator{
+			VolumeIds: volumeNameList,
+		},
+		nil,
+	)
+	if err != nil {
+		return nil, err
+	}
+	for _, volName := range volumeNameList {
+		storkVol := &storkvolume.Info{
+			VolumeID: volName,
+		}
+		for _, vol := range vols {
+			if vol.Locator.Name == volName || vol.Id == volName {
+				storkVol = p.constructStorkVolume(vol)
+				break
+			}
+		}
+		volMap[volName] = storkVol
```

Comment:
Do we want to return volumes which are not in the volDriver.Enumerate() list? I think it makes sense to move this call to just before break.

Reply:
I want to keep the behavior same as before, where we were always returning a storkvolume.Info object regardless of whether we got a successful API response for that volume from Portworx.